frontend/src/components/NavBar.tsx

```diff
@@ -0,0 +1,120 @@
+import {
+  Box,
+  Button,
+  Flex,
+  Divider,
+  Popover,
+  PopoverTrigger,
+  PopoverContent,
+  PopoverHeader,
+  PopoverBody,
+  PopoverArrow,
+  PopoverCloseButton,
+  Portal,
+  Link,
+} from "@chakra-ui/react";
+import {
+  OverViewIcon,
+  ToDoIcon,
+  PendingIcon,
+  CompletedIcon,
+  HistoryIcon,
+  SettingsIcon,
+  PlusIcon,
+} from "@/static/icons";
+import { useRouter } from "next/router";
+
+// button abstraction to include the Link and styling
+const StyleButton = ({ children, link }: { children: any; link: string }) => {
```

Comment:
@daisykucharski idk next file structure but could we put this in its own folder/file? Like either just a StyledButton.tsx or sharedComponents/StyledButton.tsx?